packages/dds/tree/src/feature-libraries/flex-map-tree/mapTreeNode.ts

```diff
@@ -407,36 +416,65 @@ class EagerMapTreeField<T extends FlexAllowedTypes> implements MapTreeField {
 	}
 
 	public context: undefined;
-}
 
-class EagerMapTreeRequiredField<T extends FlexAllowedTypes>
-	extends EagerMapTreeField<T>
-	implements FlexTreeRequiredField<T>
-{
-	public get editor(): ValueFieldEditBuilder<ExclusiveMapTree> {
-		throw unsupportedUsageError("Setting a required field");
-	}
-
-	public get content(): FlexTreeUnboxNodeUnion<T> {
-		return unboxedUnion(this.schema, this.mapTrees[0] ?? oob(), { parent: this, index: 0 });
+	// All edits to the field (i.e. mutations of the field's MapTrees) should be directed through this function.
+	// This function ensures that the parent MapTree has no empty fields (which is an invariant of `MapTree`) after the mutation.
+	protected edit(mutation: (mapTrees: ExclusiveMapTree[]) => void): void {
+		const mapTrees = this.parent.mapTree.fields.get(this.key) ?? [];
+		mutation(mapTrees);
+		if (mapTrees.length > 0) {
+			this.parent.mapTree.fields.set(this.key, mapTrees);
+		} else {
+			this.parent.mapTree.fields.delete(this.key);
+		}
 	}
 }
 
 class EagerMapTreeOptionalField<T extends FlexAllowedTypes>
 	extends EagerMapTreeField<T>
 	implements FlexTreeOptionalField<T>
 {
-	public get editor(): OptionalFieldEditBuilder<ExclusiveMapTree> {
-		throw unsupportedUsageError("Setting an optional field");
-	}
+	public editor = {
+		set: (newContent: ExclusiveMapTree | undefined) => {
+			// If the new content is a MapTreeNode, it needs to have its parent pointer updated
+			if (newContent !== undefined) {
+				nodeCache.get(newContent)?.adoptBy(this, 0);
+			}
+			// If the old content is a MapTreeNode, it needs to have its parent pointer unset
+			const oldContent = this.mapTrees[0];
+			if (oldContent !== undefined) {
+				nodeCache.get(oldContent)?.adoptBy(undefined);
+			}
+
+			this.edit((mapTrees) => {
+				if (newContent !== undefined) {
+					mapTrees[0] = newContent;
+				} else {
+					mapTrees.length = 0;
+				}
+			});
+		},
+	};
 
 	public get content(): FlexTreeUnboxNodeUnion<T> | undefined {
-		return this.mapTrees.length > 0
-			? unboxedUnion(this.schema, this.mapTrees[0] ?? oob(), {
-					parent: this,
-					index: 0,
-				})
-			: undefined;
+		const value = this.mapTrees[0];
+		if (value !== undefined) {
+			return unboxedUnion(this.schema, value, {
+				parent: this,
+				index: 0,
+			});
+		}
+
+		return undefined;
+	}
+}
+
+class EagerMapTreeRequiredField<T extends FlexAllowedTypes>
+	extends EagerMapTreeOptionalField<T>
```

Comment:
This Required class now extends the OptionalField class so that they can share the same editor implementation, since the optional-field logic works for both.

Reply:
Nit: structurally this makes sense. But semantically, it's a bit confusing. We don't export this, so it probably doesn't really matter. But if we ever made it more public, it would probably be better to introduce a shared base type and not have the optional and required types extend one another.